src/datasets/utils/streaming_download_manager.py

```diff
@@ -180,6 +180,28 @@ def xpathopen(path: Path, **kwargs):
     return xopen(_as_posix(path), **kwargs)
 
 
+def xpathglob(path, pattern):
+    """Glob function for argument of type :obj:`~pathlib.Path` that supports both local paths end remote URLs.
+
+    Args:
+        path (:obj:`~pathlib.Path`): Calling Path instance.
+        pattern (:obj:`str`): Pattern that resulting paths must match.
+
+    Yields:
+        :obj:`~pathlib.Path`
+    """
+    posix_path = _as_posix(path)
+    main_hop, *rest_hops = posix_path.split("::")
+    if is_local_path(main_hop):
+        yield from Path(main_hop).glob(pattern)
+    else:
+        fs, fs_token, globbed_paths = fsspec.get_fs_token_paths(xjoin(posix_path, pattern))
+        if "*" not in pattern:
+            globbed_paths = fs.glob(globbed_paths[0])
```

Comment:
Maybe add a comment to mention that `get_fs_token_paths` does pattern matching only if there's a `*` in the pattern ? Otherwise it could be unclear why you would need to `glob` when there's no `*` in the pattern.

```suggestion
        fs, fs_token, globbed_paths = fsspec.get_fs_token_paths(xjoin(posix_path, pattern))
        # if there's no "*" in the pattern, get_fs_token_paths() doesn't do any pattern matching
        # so to be able to glob patterns like "[0-9]", we have to call `fs.glob`
        if "*" not in pattern:
            globbed_paths = fs.glob(globbed_paths[0])
```

Reply:
I have finally refactor this so that it always calls `fs.glob` because:
- `get_fs_token_paths` does pattern matching only if there is a `*`
- The `*` in `get_fs_token_paths` only matches file names, not directory names
- `get_fs_token_paths` doesn't do pattern matching if there is `**`